packages/standard-minifier-js/plugin/minify-js.js

```diff
@@ -1,161 +1,105 @@
-      if (/^\/\/\/{6,}$/.test(sourceLine)) {
-
-        // If 4 lines back is the same exact line, we've found the framing.
-        if (contents[c - 4] === sourceLine) {
-
-          // So in that case, 2 lines back is the file path.
-          var parseErrorPath = contents[c - 2]
-            .substring(3)
-            .replace(/\s+\/\//, "");
-
-          var minError = new Error(
-            "Babili minification error " +
-            "within " + file.getPathInBundle() + ":\n" +
-            parseErrorPath +
-            (lineSrcLineNumber ? ", line " + lineSrcLineNumber : "") + "\n" +
-            "\n" +
-            lineErrorMessage + ":\n" +
-            "\n" +
-            lineSrcLineContent + "\n"
-          );
-
-          throw minError;
-        }
+    // this function tries its best to locate the original source file
+    // that the error being reported was located inside of
+    function maybeThrowMinifyErrorBySourceFile(error, file) {
+
+      const lines = file.getContentsAsString().split(/\n/);    
+      const lineContent = lines[error.line - 1];
+      
+      let originalSourceFileLineNumber = 0;
+
+      // Count backward from the failed line to find the oringal filename
+      for (let i = (error.line - 1); i >= 0; i--) {
+          let currentLine = lines[i];
+          
+          // If the line is a boatload of slashes (8 or more), we're in the right place.
+          if (/^\/\/\/{6,}$/.test(currentLine)) {
+
+              // If 4 lines back is the same exact line, we've found the framing.
+              if (lines[i - 4] === currentLine) {
+
+                  // So in that case, 2 lines back is the file path.
+                  let originalFilePath = lines[i - 2].substring(3).replace(/\s+\/\//, "");
+
+                  throw new Error(
+                      `terser minification error (${error.message})\n` + 
+                      `Source file: ${originalFilePath}  (${originalSourceFileLineNumber}:${error.col})\n` + 
+                      `Line content: ${lineContent}\n`);
+              }
+          }
+          originalSourceFileLineNumber++;
       }
     }
-  }
-
-  const toBeAdded = {
-    data: "",
-    stats: Object.create(null)
-  };
-
-  files.forEach(file => {
-    // Don't reminify *.min.js.
-    if (/\.min\.js$/.test(file.getPathInBundle())) {
-      toBeAdded.data += file.getContentsAsString();
-    } else {
-      var minified;
-
-      try {
-        minified = meteorJsMinify(file.getContentsAsString());
 
-        if (!(minified && typeof minified.code === "string")) {
-          throw new Error();
+    // this object will collect all the minified code in the 
+    // data field and post-minfiication file sizes in the stats field
+    const toBeAdded = {
+      data: "",
+      stats: Object.create(null)
+    };
+
+    files.forEach(file => {
+      // Don't reminify *.min.js.
+      if (/\.min\.js$/.test(file.getPathInBundle())) {
+        toBeAdded.data += file.getContentsAsString();
+      }
```

Comment:
Yes. Besides being pointless, for some reason Terser is very slow when minifying already-minified files.